src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -532,6 +536,20 @@ public Iterable<Article> getArticleFromSearch(String locale, String searchTerm,
         return complete(submit(req("GET", tmpl), handleList(Article.class, "results")));
     }
 
+    public MediaResponse getGuideMedias(Map<String, String> filterMap, String pageAfter, Sorting sorting) {
+        StringBuilder sb = new StringBuilder("/guide/medias");
+        List<String> statements = new ArrayList<>();
+
+        statements.add(filterMap.entrySet().stream().map(entry -> String.format("filter%s=%s", entry.getKey(), entry.getValue())).collect(Collectors.joining("&")));
+        Optional.ofNullable(pageAfter).map(it -> String.format("page[after]=%s", it)).ifPresent(statements::add);
+        Optional.ofNullable(sorting).map(it -> String.format("sort=%s", it.getSortOrder().equals(DESCENDING) ? "-".concat(it.getSortBy()) : it.getSortBy())).ifPresent(statements::add);
+        if (!statements.isEmpty()) {
+            sb.append("?").append(statements.stream().filter(a -> !a.isEmpty()).collect(Collectors.joining("&")));
+        }
+        TemplateUri tmpl = tmpl(sb.toString());
```

Comment:
maybe let's use some lib for constructing query strings. I think you would eventually need to URL encode query string values, no? like space to `%20` or `+` etc. I can imagine file name with spaces and any other characters that need to be urn encoded

Reply:
Very good point. Users definitely will include crazy characters in image filenames.

Reply:
It should be encoded in the same way as it was before. Here we just forward some params to zendesk utility class